src/main/java/org/codehaus/plexus/archiver/zip/AddedDirs.java

```diff
@@ -16,22 +16,22 @@
  */
 package org.codehaus.plexus.archiver.zip;
 
+import java.util.ArrayDeque;
+import java.util.Deque;
 import java.util.HashSet;
-import java.util.Hashtable;
 import java.util.Set;
-import java.util.Stack;
 
 /**
  * A list of directories that have been added to an archive.
  */
 public class AddedDirs
 {
 
-    private final Hashtable<String, String> addedDirs = new Hashtable<String, String>();
+    private final Set<String> addedDirs = new HashSet<String>();
 
-    public Stack<String> asStringStack( String entry )
+    public Deque<String> asStringStack(String entry )
```

Comment:
A space is missing before `String`.

But more importantly this looks like a breaking change in public method. I wonder if it is worth the risk to break existing code even it is faster. Maybe we can have separate task to remove all old collections with the next major release. Do that makes sense?